book/workflow.rst

```diff
@@ -0,0 +1,245 @@
+be applied to and the initial step that is applied when initializing the workflow for an entity.
+
+Defining the Workflow Steps
+...........................
+
+Next, you need to define the states an entity can have when the workflow is applied:
+
+.. code-block:: yaml
+    :linenos:
+
+    workflows:
+        phone_call:
+            # ...
+            steps:
+                start_call:
+                    label: 'Start Phone Call'
+                    allowed_transitions:
+                        - connected
+                        - not_answered
+                start_conversation:
+                    label: 'Call Phone Conversation'
+                    allowed_transitions:
+                        - end_conversation
+                end_call:
+                    label: 'End Phone Call'
+                    is_final: true
+
+In this example, the entity can have three states: ``start_call`` (which was configured as the
+start step above), ``start_conversation`` and ``end_call``. The ``end_call`` step is marked as
+final which means that the workflow terminates when this step is reached. The ``final`` flag does
+not mean that there cannot be any further transitions, but it is used internally for additional
+data manipulation (for example, report entities whose workflow is in a final step).
+
+The ``allowed_transitions`` defines which :ref:`transitions <book-workflow-transitions>` can be
+applied in a step.
+
+Workflow Attributes
+...................
+
+During the transition from one step to another step, the user can modify attributes of the
+underlying entity. All attributes that can be modified in any of the transitions must be enabled
+with the ``attributes`` option:
+
+.. code-block:: yaml
+    :linenos:
+
+    workflows:
+        phone_call:
+            # ...
+            attributes:
+                phone_call:
+                    label: Phone Call
+                    type: entity
+                    options:
+                        class: Acme\Bundle\DemoWorkflowBundle\Entity\PhoneCall
+                call_timeout:
+                    type: integer
+                    label: 'Call Timeout'
+                call_successfull:
+                    type: boolean
+                    label: 'Call Successful'
+                conversation_successful:
+                    type: boolean
+                    label: 'Conversation Successful'
+                conversation_comment:
+                    type: string
+                    label: 'Conversation Comment'
+                conversation_result:
+                    type: string
+                    label: 'Conversation Result'
+                conversation:
+                    type: entity
+                    label: 'Conversation'
+                    options:
+                        class: Acme\Bundle\DemoWorkflowBundle\Entity\PhoneConversation
+
+.. tip::
+
+    This example assumes that the keys used under the ``attributes`` option are the names of the
```

Comment:
Thanks for making the description better fit the actual code. I opened #193 with a proposal of how to tweak the description a bit more. Let me know what you think about it.